ecosystem/indexer/src/models/transactions.rs

```diff
@@ -32,18 +38,105 @@ pub struct Transaction {
 }
 
 impl Transaction {
-    pub fn get_by_hash(transaction_hash: String, connection: &PgPoolConnection) {
-        let transactions = transactions::table.load::<Transaction>(connection)?;
-        let user_transactions = UserTransaction::belonging_to(&transactions)
-            .load::<UserTransaction>(&connection)?
-            .grouped_by(&users);
-        let block_metadata_transactions = BlockMetadataTransaction::belonging_to(&transactions)
-            .load::<BlockMetadataTransaction>(&connection)?
-            .grouped_by(&users);
-        (transactions, user_transactions, block_metadata_transactions)
+    pub fn get_many_by_version(
+        start_version: i64,
+        number_to_get: i64,
+        connection: &PgPoolConnection,
+    ) -> diesel::QueryResult<
+        Vec<(
+            Transaction,
+            Option<UserTransaction>,
+            Option<BlockMetadataTransaction>,
+        )>,
+    > {
+        let mut transactions = transactions::table
+            .filter(transactions::version.ge(start_version))
+            .limit(number_to_get)
+            .load::<Transaction>(connection)?;
+
+        let mut user_transactions: Vec<Vec<UserTransaction>> =
+            UserTransaction::belonging_to(&transactions)
+                .load::<UserTransaction>(connection)?
+                .grouped_by(&transactions);
+
+        let mut block_metadata_transactions: Vec<Vec<BlockMetadataTransaction>> =
+            BlockMetadataTransaction::belonging_to(&transactions)
+                .load::<BlockMetadataTransaction>(connection)?
+                .grouped_by(&transactions);
+
+        // Convert to the nice result tuple
+        let mut result = vec![];
+        while !transactions.is_empty() {
+            result.push((
+                transactions.pop().unwrap(),
+                user_transactions.pop().unwrap().pop(),
+                block_metadata_transactions.pop().unwrap().pop(),
+            ))
+        }
+
+        Ok(result)
     }
 
-    pub fn get_by_version(version: u64) {}
+    pub fn get_by_version(
+        version: u64,
+        connection: &PgPoolConnection,
+    ) -> diesel::QueryResult<(
+        Transaction,
+        Option<UserTransaction>,
+        Option<BlockMetadataTransaction>,
+    )> {
+        let transaction = transactions::table
+            .filter(transactions::version.eq(version as i64))
+            .first::<Transaction>(connection)?;
+
+        let (user_transaction, block_metadata_transaction) =
+            transaction.get_details_for_transaction(connection)?;
+
+        Ok((transaction, user_transaction, block_metadata_transaction))
+    }
+
+    pub fn get_by_hash(
+        transaction_hash: &str,
+        connection: &PgPoolConnection,
+    ) -> diesel::QueryResult<(
+        Transaction,
+        Option<UserTransaction>,
+        Option<BlockMetadataTransaction>,
```

Comment:
Why not build a new transaction enum for the indexer or convert back into an API transaction?

Reply:
I could I guess, yeah